### PR TITLE
Commit

### PR DESCRIPTION
**Pull request by Sasha192**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +29,21 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java:32]: add other checking for equals

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +29,21 @@ public Integer getLength() {
...
+            return ((Rectangle) obj).getColor().equals(this.color)
+                    && ((Rectangle) obj).getLength().equals(this.length)
+                    && ((Rectangle) obj).getWidth().equals(this.width);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.color,this.length,this.width);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        return Objects.hash(this.color, this.length, this.width);
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package core.basesyntax;
 
+import java.util.Objects;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @Sasha192 lets not use Objects library

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,24 @@ public Integer getLength() {
...
+            return ((Rectangle) obj).getColor().equals(this.color)
+                    && ((Rectangle) obj).getLength().equals(this.length)
+                    && ((Rectangle) obj).getWidth().equals(this.width);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return java.util.Objects.hash(this.color, this.length, this.width);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @Sasha192 do not use Objects class

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,24 @@ public Integer getLength() {
...
+    }
+
+
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: empty lines

**Pull request closed without merging**